extensions/parquet/table/src/main/java/io/deephaven/parquet/table/location/ParquetColumnResolverMap.java

```diff
@@ -0,0 +1,63 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.parquet.table.location;
+
+import io.deephaven.annotations.BuildableStyle;
+import org.apache.parquet.column.ColumnDescriptor;
+import org.apache.parquet.schema.MessageType;
+import org.immutables.value.Value;
+
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * A {@link ParquetColumnResolver} implementation based on a map from Deephaven column names to Parquet
+ * {@link ColumnDescriptor column descriptors}.
+ */
+@Value.Immutable
+@BuildableStyle
+public abstract class ParquetColumnResolverMap implements ParquetColumnResolver {
+
+    public static Builder builder() {
+        return ImmutableParquetColumnResolverMap.builder();
+    }
+
+    /**
+     * The Parquet schema.
+     */
+    public abstract MessageType schema();
+
+    /**
+     * The map from Deephaven column name to {@link ColumnDescriptor}. The {@link #schema()} must contain each column
+     * descriptor.
+     */
+    public abstract Map<String, ColumnDescriptor> mapping();
```

Comment:
Do we need this to keep `ColumnDescriptor` as part of the interface of the implementation? Do we want the Iceberg implementation using this?

Reply:
I'm very much in favor of keeping this implementation strongly-typed with the safety checks. Iceberg should be able to use this implementation (and even use the same Factory to create it) in some situations.

Reply:
Changed my mind; the map implementation is now very generic, no Parquet specific types.